src/Dommel/DommelMapper.cs

```diff
@@ -868,6 +868,78 @@ private static string BuildSelectSql<TEntity>(Expression<Func<TEntity, bool>> pr
         }
 
         /// <summary>
+        /// Selects the paginated entities matching the specified predicate.
+        /// </summary>
+        /// <typeparam name="TEntity">The type of the entity.</typeparam>
+        /// <param name="connection">The connection to the database. This can either be open or closed.</param>
+        /// <param name="predicate">A predicate to filter the results.</param>
+        /// <param name="orderByExpression">A expression to order by the results.</param>
+        /// <param name="orderByAsc">Order by type.</param>
+        /// <param name="pageNo">The page number of the results.</param>
+        /// <param name="pageSize">The page size of the results.</param>
+        /// <returns>
+        /// A collection of entities of type <typeparamref name="TEntity"/> matching the specified
+        /// <paramref name="predicate"/>.
+        /// </returns>
+        public static IEnumerable<TEntity> Query<TEntity>(this IDbConnection connection,
+            Expression<Func<TEntity, bool>> predicate,
+            Expression<Func<TEntity, object>> orderByExpression = null,
+            bool orderByAsc = true,
+            int? pageNo = null, int? pageSize = null)
+        {
+            var tableName = Resolvers.Table(typeof(TEntity));
+            var orderBySql = BuildOrderSql(orderByExpression, orderByAsc);
+
+            DynamicParameters parameters;
+            var sql = BuildSelectSql<TEntity>(predicate, out parameters);
+            if (pageNo == null || pageSize == null)
+            {
+                sql = $"{sql} {orderBySql}";
+            }
+            else
+            {
+                sql = BuildPaginationSql(connection, tableName, sql, orderBySql, orderByAsc, pageNo.Value, pageSize.Value);
+            }
+
+            return connection.Query<TEntity>(sql, (object) parameters);
+        }
+
+        private static string BuildOrderSql<TEntity>(Expression<Func<TEntity, object>> predicate, bool orderByAsc)

```

Comment:
Is it possible to implement this method in the `SqlExpression` class? Seems a better fit to me and doesn't require stuff like `VisitMemberAccess` to be public.